src/spdx_tools/spdx/datetime_conversions.py

```diff
@@ -16,4 +17,11 @@ def datetime_to_iso_string(date: datetime) -> str:
     """
     Return an ISO-8601 representation of a datetime object.
     """
+    if date.microsecond != 0:
+        date = date.replace(microsecond=0)
+        logging.warning(
```

Comment:
I'm not 100% convinced this warrants a warning. Since there is no dedicated datetime type without microsecond precision and the most typical way to create a datetime object will add microsecond precision, as a user it might be annoying to deal with those warnings.
But I'm not strictly opposed, either.

Reply:
You are probably correct, I removed it.